bluzelle/bluzelle-js
# KEP-1503 Downgrade Web3 version

QA Travis build is seeing this error 
```
TypeError: Cannot read property 'HttpProvider' of undefined
      at bluzelle (webpack:///./src/main.js?:26:52)
      at initializeClient (src/clientManager.js:4:24)
      at Context.<anonymous> (tests/shared/hooks.js:43:28)
      at process._tickCallback (internal/process/next_tick.js:68:7)
```

I've replicated it locally and determined that it's caused by using different Web3 versions in the same project. I've resolved it by downgrading bluzelle-js to 1.0.0-beta.37 and using this version across bluzelleESR, qa and bluzelle-js. 

BluzelleESR and qa both use the truffle framework or its core modules, which is locked to 1.0.0-beta.37.

~After downgrading to .37, there's a webpack error that does not seem to hinder the functionality of the client.~

**Resolved webpack error by ignoring electron package optionally required by sub dependency. See https://github.com/sindresorhus/got/issues/345** 

```
Executing pre-build scripts
(node:90387) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
Hash: a6de972783673f8671ffabdb253b4b9dc86cda43
Version: webpack 4.21.0
Child
    Hash: a6de972783673f8671ff
    Time: 3516ms
    Built at: 06/10/2019 4:06:37 PM
               Asset      Size  Chunks             Chunk Names
    bluzelle-node.js  3.96 MiB    main  [emitted]  main
    Entrypoint main = bluzelle-node.js
    [./BluzelleESR/build/contracts/BluzelleESR.json] 678 KiB {main} [built]
    [./package.json] 1.04 KiB {main} [built]
    [./src/main.js] 4.31 KiB {main} [built]
    [./src/swarmClient/1_connection_layer.js] 10.6 KiB {main} [built]
    [./src/swarmClient/2_serialization_layer.js] 1.49 KiB {main} [built]
    [./src/swarmClient/3_crypto_layer.js] 3.78 KiB {main} [built]
    [./src/swarmClient/4_collation_layer.js] 4.17 KiB {main} [built]
    [./src/swarmClient/5_broadcast_layer.js] 3.96 KiB {main} [built]
    [./src/swarmClient/6_redirect_layer.js] 1.61 KiB {main} [built]
    [./src/swarmClient/7_envelope_layer.js] 2.01 KiB {main} [built]
    [./src/swarmClient/8_metadata_layer.js] 3.04 KiB {main} [built]
    [./src/swarmClient/9_api_layer.js] 17.6 KiB {main} [built]
    [./src/swarmClient/ecdsa_secp256k1.js] 5.33 KiB {main} [built]
    [./src/swarmClient/main.js] 4.41 KiB {main} [built]
    [assert] external "assert" 42 bytes {main} [built]
        + 715 hidden modules

    ERROR in ./node_modules/got/index.js
    Module not found: Error: Can't resolve 'electron' in '/Users/johnlam/Work/tmp/qa-travis-debug-v1/bluzelle-js/node_modules/got'
     @ ./node_modules/got/index.js 45:20-39
     @ ./node_modules/swarm-js/lib/files.js
     @ ./node_modules/swarm-js/lib/api-node.js
     @ ./node_modules/web3-bzz/src/index.js
     @ ./node_modules/web3/src/index.js
     @ ./src/main.js
Child
    Hash: abdb253b4b9dc86cda43
    Time: 3597ms
    Built at: 06/10/2019 4:06:38 PM
              Asset      Size  Chunks             Chunk Names
    bluzelle-web.js  3.43 MiB    main  [emitted]  main
    Entrypoint main = bluzelle-web.js
    [./BluzelleESR/build/contracts/BluzelleESR.json] 678 KiB {main} [built]
    [./package.json] 1.04 KiB {main} [built]
    [./src/main.js] 4.31 KiB {main} [built]
    [./src/swarmClient/1_connection_layer.js] 10.6 KiB {main} [built]
    [./src/swarmClient/2_serialization_layer.js] 1.49 KiB {main} [built]
    [./src/swarmClient/3_crypto_layer.js] 3.78 KiB {main} [built]
    [./src/swarmClient/4_collation_layer.js] 4.17 KiB {main} [built]
    [./src/swarmClient/5_broadcast_layer.js] 3.96 KiB {main} [built]
    [./src/swarmClient/6_redirect_layer.js] 1.61 KiB {main} [built]
    [./src/swarmClient/7_envelope_layer.js] 2.01 KiB {main} [built]
    [./src/swarmClient/8_metadata_layer.js] 3.04 KiB {main} [built]
    [./src/swarmClient/9_api_layer.js] 17.6 KiB {main} [built]
    [./src/swarmClient/ecdsa_secp256k1.js] 5.33 KiB {main} [built]
    [./src/swarmClient/log.js] 1.78 KiB {main} [built]
    [./src/swarmClient/main.js] 4.41 KiB {main} [built]
        + 450 hidden modules
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @ setup-bluzelle-js: `git submodule update --init --recursive; cd bluzelle-js && npm install && printf "qa repo package-lock.json:" && cat package-lock.json && cd bluzelleESR && npm install && printf "qa repo package-lock.json:" && cat package-lock.json && npx truffle compile && cd .. && webpack`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the @ setup-bluzelle-js script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

```
